idaes/apps/uncertainty_propagation/uncertainties.py

```diff
@@ -0,0 +1,511 @@
+        build the objective function
+    theta_names: list of strings
+        List of Var names to estimate
+    obj_function: function, optional
+        Function used to formulate parameter estimation objective, 
+        generally sum of squared error between measurements and model variables.    
+    tee: bool, optional
+        Indicates that ef solver output should be teed
+    diagnostic_mode: bool, optional
+        If True, print diagnostics from the solver
+    solver_options: dict, optional
+        Provides options to the solver (also the name of an attribute)
+    
+    Returns
+    -------
+    results : namedtuple
+        results.obj: float
+            objective function value for the given obj_function 
+        results.theta_out: dictionary
+            Estimated parameters
+        results.cov: numpy.ndarray
+            Covariance of theta_out
+        results.gradient_f_dic: numpy.ndarray 
+            gradient of the objective function with respect to all decision variables at the optimal solution 
+            with variable name as key e.g) dic = {d(f)/d(x1):0.1, d(f)/d(x2):0.1}
+        results.gradient_c_dic: numpy.ndarray
+            gradient of the constraints with respect to all decision variables at the optimal solution
+            with constraint number and variable name as key e.g) dic = {d(c1)/d(x1):1.1, d(c4)/d(x2):0.1}
+            Only non-zero gradients are included.
+        results.dsdp_dic: dict
+            gradient vector of the (decision variables, parameters) with respect to paramerters (=theta_name).
+            e.g) dict = {'d(x1)/d(p1)': 1.0, 'd(x2)/d(p1)': 0.0, 'd(p1)/d(p1)': 1.0, 'd(p2)/d(p1)': 0.0, 
+                     'd(x1)/d(p2)': 0.0, 'd(x2)/d(p2)': 1.0, 'd(p1)/d(p2)': 0.0, 'd(p2)/d(p2)': 1.0}
+        results.propagation_f: dict
+            error propagation in the objective function with the dictionary key, 'objective' 
+            if the objective function doesn't include any uncertain parameters, return an empty dictionary        
+        results.propagation_c: dict
+            error propagation in the constraints with the dictionary keys, 'constraints l'
+            where l is the line number. 
+            if no constraint includes uncertain parameters, return an empty dictionary      
+        
+    Raises
+    ------
+    ValueError
+        When tee entry is not Boolean
+    ValueError
+        When diagnostic_mode entry is not Boolean
+    ValueError
+        When solver_options entry is not None and a Dictionary
+    Warnings
+        When an element of theta_names includes a space
+    """
+    if not isinstance(tee, bool):
+        raise ValueError('tee  must be boolean.')
+    if not isinstance(diagnostic_mode, bool):
+        raise ValueError('diagnostic_mode  must be boolean.')    
+    if not solver_options==None:
+        if not isinstance(solver_options, dict):
+            raise ValueError('solver_options must be dictionary.')
```

Comment:
This is TypeError

Reply:
@JanghoPark-LBL This still needs to be fixed.